File: ai-service/app/models/temp_business_data_sample.py
from sqlalchemy import Column, BigInteger, Integer, Float, String, Date, Boolean, Text, DateTime
from app.database import Base
import datetime

class TempBusinessDataSample(Base):
    __tablename__ = "temp_business_data_sample"

    id = Column(BigInteger, primary_key=True, index=True)

    dataset_id = Column(BigInteger, nullable=False)

    product = Column(String, nullable=True)
    category = Column(String, nullable=True)

    quantity = Column(Integer, nullable=True)
    price = Column(Float, nullable=True)
    total = Column(Float, nullable=True)

    date = Column(Date, nullable=True)
    month = Column(Integer, nullable=True)
    year = Column(Integer, nullable=True)

    is_valid = Column(Boolean, default=False)
    validation_error = Column(Text, nullable=True)
    error_level = Column(String, nullable=True)

    suggested_date = Column(Date, nullable=True)
    suggested_month = Column(Integer, nullable=True)
    suggested_year = Column(Integer, nullable=True)
    suggested_category = Column(String, nullable=True)
    suggested_product = Column(String, nullable=True)
    suggested_price = Column(Float, nullable=True)
    suggested_quantity = Column(Float, nullable=True)
    suggested_total = Column(Float, nullable=True)

    user_confirmed = Column(Boolean, default=False)
    ai_correction = Column(Boolean, default=False)
    corrected_field_by_ai = Column(Text, nullable=True)
    corrected_data = Column(Text, nullable=True)

    created_at = Column(DateTime, default=datetime.datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.datetime.utcnow)

File: laravel-backend/app/Models/TempBusinessDataSample.php
<?php
namespace App\Models;
use Illuminate\Database\Eloquent\Model;

class TempBusinessDataSample extends Model
{
    protected $table = 'temp_business_data_sample';
    protected $fillable = [
        'dataset_id',
        'date',
        'month',
        'year',
        'product',
        'category',
        'quantity',
        'price',
        'total',
        'is_valid',
        'validation_errors',
        'error_level',
        'suggested_date',
        'suggested_month',
        'suggested_year',
        'suggested_category',
        'suggested_product',
        'suggested_price',
        'suggested_quantity',
        'suggested_total',
        'user_confirmed',
        'ai_correction',
        'corrected_Field_by_ai',
        'corrected_data'
    ];

    protected $casts = [
        'is_valid'          => 'boolean',
        'validation_errors' => 'array',       // json column — auto encode/decode
        'user_confirmed'    => 'boolean',
        'ai_correction'     => 'boolean',
        'suggested_date'    => 'datetime',
        'suggested_month'   => 'integer',
        'suggested_year'    => 'integer',
        'suggested_price'   => 'float',
        'suggested_quantity'=> 'float',
        'suggested_total'   => 'float',
    ];

    public function dataset()
    {
        return $this->belongsTo(Dataset::class, 'dataset_id');
        
    }

     public function productCorrection()
    {
        return $this->hasOne(AiCorrection::class, 'original_value', 'product')
                    ->where('column_name', 'product');
    }

    public function categoryCorrection()
    {
        return $this->hasOne(AiCorrection::class, 'original_value', 'category')
                    ->where('column_name', 'category');
    }
}
?>